python/web/index.py

```diff
@@ -0,0 +1,202 @@
+import os
+import flask
+import gevent.pywsgi
+import subprocess
+import sys
+from io import StringIO
+from io import TextIOWrapper, BytesIO
+
+import examples
+# from .list import create_list
+
+app = flask.Flask(__name__)
+
+
+@app.route('/python/')
+def root():
+    return create_list()
+
+
+def file_get_contents(example):
+    current_example = os.getcwd() + '/examples/{0}.py'.format(example)
+
+    text = open(current_example, "r")
+    contents = text.read()
+    text.close()
+    return contents
+
+
+def capture_output(example):
+
+    file = os.getcwd() + '/examples/' + example + '.py'
+
+    # # setup the environment
+    # old_stdout = sys.stdout
+    # sys.stdout = TextIOWrapper(BytesIO(), sys.stdout.encoding)
+    #
+    # # do some writing (indirectly)
+    # # write("blub")
+    # os.system('python ' + file)
+    #
+    # # get output
+    # sys.stdout.seek(0)  # jump to the start
+    # out = sys.stdout.read()  # read output
+    #
+    # # restore stdout
+    # sys.stdout.close()
+    # sys.stdout = old_stdout
+
+    cmd = 'python3 ' + file
+    # out = subprocess.check_output(cmd, shell=Truerue)
+    # out = subprocess.run(cmd, stdout=subprocess.PIPE)
+    p = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+    (out, err) = p.communicate()
+
+    # file = os.getcwd() + '/examples/' + example + '.py'
+    #
+    # # setup the environment
+    # backup = sys.stdout
+    #
+    # # Capture the output
+    # sys.stdout = StringIO()
+    #
+    # os.system('python ' + file)
+    #
+    # # release output
+    # # out = sys.stdout.getvalue()
+    # out = backup.getvalue()
+
+    return out
+
+
+@app.route('/python/<example>')
+def add_example_route(example):
```

Comment:
Should route action functions begin with `add_`?  That seems like an odd convention.  @NickAnderegg ?

Reply:
I don't have a strong opinion on this one. I haven't seen many conventions. We could go with `create_example_route`? `generate_example_route`?